static/absolute/js/searchPage.js

```diff
@@ -41,54 +54,67 @@ window.onload = function () {
           location: [lat, long],
         }),
       });
-      data = await resp.json();
+      postResponse = await resp.json();
     } catch (err) {
       console.log(err);
       alert("error posting to api");
       return;
     }
 
-    if (data.status !== 200) {
+    if (postResponse.status !== 200) {
       // TODO(ved): How should we display errors?
-      console.log(data.error);
+      console.log(postResponse.error);
       alert("error posting to api");
       return;
     }
 
+    //Data was submitted successfully. We send a message to the server to then update each client with the same event ID.
+    socket.emit("data submitted", eventId);
```

Comment:
I'm curious why this is necessary - the server should already know that you updated the info (you POSTed to api/$eventId), why does it require another message to know that if should be updating the other clients?

Reply:
Removed. The server now emits a refresh message in the POST method.